src/public/media.ts

```diff
@@ -11,395 +11,400 @@ import {
+   * Launch camera, capture image or choose image from gallery and return the images as a File[] object to the callback.
+   * Callback will be called with an error, if there are any. App should first check the error.
+   * If it is present the user can be updated with appropriate error message.
+   * If error is null or undefined, then files will have the required result.
+   * Note: Currently we support getting one File through this API, i.e. the file arrays size will be one.
+   * Note: For desktop, this API is not supported. Callback will be resolved with ErrorCode.NotSupported.
+   * @see File
+   * @see SdkError
    */
-  public getMedia(callback: (error: SdkError, blob: Blob) => void): void {
+  export function captureImage(callback: (error: SdkError, files: File[]) => void): void {
     if (!callback) {
-      throw new Error('[get Media] Callback cannot be null');
+      throw new Error('[captureImage] Callback cannot be null');
     }
     ensureInitialized(FrameContexts.content, FrameContexts.task);
-    if (!isAPISupportedByPlatform(mediaAPISupportVersion)) {
-      const oldPlatformError: SdkError = { errorCode: ErrorCode.OLD_PLATFORM };
-      callback(oldPlatformError, null);
+
+    if (!GlobalVars.isFramelessWindow) {
+      const notSupportedError: SdkError = { errorCode: ErrorCode.NOT_SUPPORTED_ON_PLATFORM };
+      callback(notSupportedError, undefined);
       return;
     }
-    if (!validateGetMediaInputs(this.mimeType, this.format, this.content)) {
-      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
-      callback(invalidInput, null);
+
+    if (!isAPISupportedByPlatform(captureImageMobileSupportVersion)) {
+      const oldPlatformError: SdkError = { errorCode: ErrorCode.OLD_PLATFORM };
+      callback(oldPlatformError, undefined);
       return;
     }
-    const actionName = generateGUID();
-    const helper: MediaHelper = {
-      mediaMimeType: this.mimeType,
-      assembleAttachment: [],
-    };
-    const params = [actionName, this.content];
-    this.content && callback && sendMessageRequestToParent('getMedia', params);
-    function handleGetMediaRequest(response: string): void {
-      if (callback) {
-        const mediaResult: MediaResult = JSON.parse(response);
-        if (mediaResult.error) {
-          callback(mediaResult.error, null);
-          GlobalVars.handlers['getMedia' + actionName] = null;
-        } else {
-          if (mediaResult.mediaChunk) {
-            // If the chunksequence number is less than equal to 0 implies EOF
-            // create file/blob when all chunks have arrived and we get 0/-1 as chunksequence number
-            if (mediaResult.mediaChunk.chunkSequence <= 0) {
-              const file = createFile(helper.assembleAttachment, helper.mediaMimeType);
-              callback(mediaResult.error, file);
-              GlobalVars.handlers['getMedia' + actionName] = null;
+
+    const messageId = sendMessageRequestToParent('captureImage');
```

Comment:
We need to update this name also with prefix as media "media.captureImage" . Please do it for all APIs in this file. On mobile side to make it backward compatible, We have to support both "captureImage" and "media.captureImage" for some time.

Reply:
If we do that it will be a breaking change for the SDK though and the new version wouldn't be usable on the old mobile clients that haven't been updated with this change. I think it might be safest to leave it as is if any partners have already taken a dependency on these APIs.

Reply:
sounds good. In that case we can keep existing APIs name as it is and we can put media prefix for all new APIs that we add in future.